src/BackupChain.cs

```diff
@@ -66,54 +75,35 @@ private static IEnumerable<BackupMetadata> MostRecentFullBackup(IEnumerable<Back
+      return fullBackupsOrdered.First();
     }
 
-    private static IEnumerable<BackupMetadata> MostRecentDiffBackup(IEnumerable<BackupMetadata> backups,
+    private static StripedBackupSet MostRecentDiffBackup(IEnumerable<StripedBackupSet> backups,
       BackupMetadata lastFullBackup)
     {
-      var diffBackupsOrdered = backups
-        .Where(b =>
-                 b.BackupType == BackupFileTools.BackupType.Diff &&
-                 b.DatabaseBackupLsn == lastFullBackup.CheckpointLsn)
-        .OrderByDescending(b => b.LastLsn).ToList();
-
-      if(!diffBackupsOrdered.Any())
-        return new List<BackupMetadata>();
-
-      var targetLastLsn = diffBackupsOrdered.First().LastLsn;
-      // get all the stripes of this backup
-      return diffBackupsOrdered.Where(diffBackup => diffBackup.LastLsn == targetLastLsn);
+      return backups.Where(b => b.BackupType == BackupFileTools.BackupType.Diff &&
+                                b.DatabaseBackupLsn == lastFullBackup.CheckpointLsn)
+                    .OrderByDescending(b => b.LastLsn)
+                    .FirstOrDefault();
     }
 
-    private static IEnumerable<BackupMetadata> NextLogBackup(IEnumerable<BackupMetadata> backups,
-      BackupMetadata prevBackup)
+    private static StripedBackupSet NextLogBackup(IEnumerable<StripedBackupSet> backups,
+      BackupMetadata prevLog)
     {
-     // also gets all the stripes of the next backup
-     return backups.Where(b => b.BackupType == BackupFileTools.BackupType.Log &&
-                               prevBackup.LastLsn == b.FirstLsn &&
-                               !StripedBackupEqualityComparer.Instance.Equals(prevBackup, b));
+      return backups.SingleOrDefault(b => b.BackupType == BackupFileTools.BackupType.Log &&
+                                          prevLog.LastLsn == b.FirstLsn);
     }
 
-    private static IEnumerable<BackupMetadata> FirstLogBackupInChain(IEnumerable<BackupMetadata> backups, 
-     BackupMetadata lastDiff)
+    private static StripedBackupSet FirstLogInChain(IEnumerable<StripedBackupSet> backups, 
+     BackupMetadata lastBackup)
     {
       var possibleLogs = backups.Where(b => b.BackupType == BackupFileTools.BackupType.Log &&
-                                       lastDiff.LastLsn >= b.FirstLsn &&
-                                       lastDiff.LastLsn <= b.LastLsn &&
-                                       !StripedBackupEqualityComparer.Instance.Equals(lastDiff, b));
-
-      var striped = StripedBackupSet.GetStripedBackupSetChain(possibleLogs);
+                                       lastBackup.LastLsn >= b.FirstLsn &&
+                                       lastBackup.LastLsn <= b.LastLsn).ToList();
 
-      if (striped.Count() > 1)
-      {
-       return striped.OrderBy(striped => striped.StripedBackups.First().LastLsn).Last().StripedBackups;
-      }
-
-      return possibleLogs;
-  }
+      return possibleLogs.OrderByDescending(b => b.LastLsn).FirstOrDefault();

```

Comment:
Re-reading this, I think it can be combined with the above LINQ.
There isn't anything that complicated we are doing in this second LINQ.

Combining them also means we are removing the `ToList()` so it stays as an IEnumerable until the `FirstOrDefault`.

(Also once again, `Where()` can be removed)

Reply:
It could be combined, but I personally prefer keeping the methods separate. In `NextLogBackup` the query should only match a single backup or none at all, which can be enforced through the use of `SingleOrDefault()`. The rule in `FirstLogInChain` isn't as strict since it could potentially lead to the edge-case encountered in https://github.com/factset/AgDatabaseMove/pull/109.

Reply:
> but I personally prefer keeping the methods separate

ah, no - looks like there has been a misunderstanding here.

I'm not referring to combining the functions. I'm taking about combining the LINQ in this `FirstLogInChain()` function.
...but looks like you have done that now 👍🏽